src/crimson/os/seastore/journal/segmented_journal.cc

```diff
@@ -312,8 +312,8 @@ SegmentedJournal::replay_segment(
 	cursor,
 	header.segment_nonce,
 	std::numeric_limits<size_t>::max(),
-	dhandler).safe_then([](auto){}
-      ).handle_error(
+	dhandler
+      ).discard_result().handle_error(
```

Comment:
could you please check why we return a value need to be discarded?  for e.g. scan_valid_records, seems every place call it will discard the return value.  so for this function, we needn't return a value, right?

Reply:
After checking the "scan_valid_records" and related functions, I determined that this return value is used to count the length of the header, it's only for log output and unrelated to reply process. Seems we can remove it savely.